### PR TITLE
fix: [ANDROAPP-5376 ] Login button doesn't work and doesn't present any error

### PR DESCRIPTION
**Pull request by xavimolloy**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] 11.X - 13.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/login/LoginViewModel.kt [hunk 1]
```diff
@@ -178,11 +178,16 @@ class LoginViewModel(
     }
 
     fun onLoginButtonClick() {
-        view.hideKeyboard()
-        analyticsHelper.setEvent(LOGIN, CLICK, LOGIN)
-        increment()
-        logIn()
-        decrement()
+        try {
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/login/LoginViewModel.kt:181]: Why don't you create a couple of tests for this?. For success and error case.

**Review by andresmr — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/dhis2/usescases/login/LoginViewModel.kt (hunk 1 above)
Comment (xavimolloy) [reply app/src/main/java/org/dhis2/usescases/login/LoginViewModel.kt:181]: Yes, some were already active but I have linked the other issue we had for the ignored ones.

Comment (andresmr) [reply app/src/main/java/org/dhis2/usescases/login/LoginViewModel.kt:181]: I was thinking about the LoginViewModel as a unit tests

File: app/src/androidTest/java/org/dhis2/usescases/login/LoginRobot.kt [hunk 2]
```diff
@@ -104,7 +101,7 @@ class LoginRobot : BaseRobot() {
             CoreMatchers.allOf(
                 hasExtra(
                     WebViewActivity.WEB_VIEW_URL,
-                    "${BaseTest.MOCK_SERVER_URL}/dhis-web-commons/security/recovery.action"
+                    "${LoginTest.MOCK_SERVER_URL_FOR_ACCOUNT_RECOVERY_TEST}/dhis-web-commons/security/recovery.action"
```
Comment (andresmr) [inline app/src/androidTest/java/org/dhis2/usescases/login/LoginRobot.kt:104]: I think it worked with the previous URL

File: app/src/androidTest/java/org/dhis2/usescases/login/LoginTest.kt [hunk 3]
```diff
@@ -96,10 +97,9 @@ class LoginTest : BaseTest() {
     fun shouldLaunchWebViewWhenClickAccountRecoveryAndServerIsFilled() {
         enableIntents()
         startLoginActivity()
-
         loginRobot {
             clearServerField()
-            typeServer(MOCK_SERVER_URL)
+            typeServer(MOCK_SERVER_URL_FOR_ACCOUNT_RECOVERY_TEST)
```
Comment (andresmr) [inline app/src/androidTest/java/org/dhis2/usescases/login/LoginTest.kt:102]: I think there is no need to change the URL

**Review by andresmr — CHANGES_REQUESTED** [review]

File: app/src/test/java/org/dhis2/usescases/login/LoginViewModelTest.kt [hunk 4]
```diff
@@ -62,13 +64,12 @@ class LoginPresenterTest {
     private val analyticsHelper: AnalyticsHelper = mock()
     private val crashReportController: CrashReportController = mock()
     private val network: NetworkUtils = mock()
+    private lateinit var loginViewModel: LoginViewModel
+    private lateinit var loginViewModelWithNullUserManager: LoginViewModel
```
Comment (andresmr) [inline app/src/test/java/org/dhis2/usescases/login/LoginViewModelTest.kt:68]: I think that you don't need two variables as is going to be assigned once per test, you can reuse loginViewModel

**Review by andresmr — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Review by mmmateos — APPROVED** [review]

**Pull request merged**